staging/src/k8s.io/dynamic-resource-allocation/client/generic.go

```diff
@@ -0,0 +1,332 @@
+	apis := newCall(t.c, func(currentAPI int32) (*N, error) {
+		switch currentAPI {
+		case useV1beta1API:
+			return nil, t.v1beta1.DeleteCollection(ctx, opts, listOpts)
+		default:
+			return nil, t.native.DeleteCollection(ctx, opts, listOpts)
+		}
+	})
+	_, err := apis.run()
+	return err
+}
+
+func (t *typedClient[NP, N, NL, NAC, OP, O, OL, OAC]) Get(ctx context.Context, name string, opts metav1.GetOptions) (*N, error) {
+	apis := newCall(t.c, func(currentAPI int32) (*N, error) {
+		switch currentAPI {
+		case useV1beta1API:
+			return getWithConversion[N, O](func() (*O, error) {
+				return t.v1beta1.Get(ctx, name, opts)
+			})
+		default:
+			return t.native.Get(ctx, name, opts)
+		}
+	})
+	return apis.run()
+}
+
+func (t *typedClient[NP, N, NL, NAC, OP, O, OL, OAC]) List(ctx context.Context, opts metav1.ListOptions) (*NL, error) {
+	apis := newCall(t.c, func(currentAPI int32) (*NL, error) {
+		switch currentAPI {
+		case useV1beta1API:
+			return getWithConversion[NL, OL](func() (*OL, error) {
+				return t.v1beta1.List(ctx, opts)
+			})
+		default:
+			return t.native.List(ctx, opts)
+		}
+	})
+	return apis.run()
+}
+
+func (t *typedClient[NP, N, NL, NAC, OP, O, OL, OAC]) Watch(ctx context.Context, opts metav1.ListOptions) (watch.Interface, error) {
+	apis := newCall(t.c, func(currentAPI int32) (watch.Interface, error) {
+		switch currentAPI {
+		case useV1beta1API:
+			return watchWithConversion[NP, N, OP](func() (watch.Interface, error) {
+				return t.v1beta1.Watch(ctx, opts)
+			})
+		default:
+			return t.native.Watch(ctx, opts)
+		}
+	})
+	return apis.run()
+}
+
+func (t *typedClient[NP, N, NL, NAC, OP, O, OL, OAC]) Patch(ctx context.Context, name string, pt types.PatchType, data []byte, opts metav1.PatchOptions, subresources ...string) (result *N, err error) {
+	return nil, ErrNotImplemented
+}
+
+func (t *typedClient[NP, N, NL, NAC, OP, O, OL, OAC]) Apply(ctx context.Context, obj *NAC, opts metav1.ApplyOptions) (result *N, err error) {
+	return nil, ErrNotImplemented
+}
+
+func (t *typedClient[NP, N, NL, NAC, OP, O, OL, OAC]) ApplyStatus(ctx context.Context, obj *NAC, opts metav1.ApplyOptions) (result *N, err error) {
+	return nil, ErrNotImplemented
+}
+
+func newCall[N any](c *client, call func(currentAPI int32) (N, error)) callSequence[N] {
+	seq := callSequence[N]{
+		c:    c,
+		call: call,
+	}
+	return seq
+}
+
+type callSequence[N any] struct {
+	c    *client
+	call func(currentAPI int32) (N, error)
+}
+
+func (s *callSequence[N]) run() (N, error) {
+	currentAPI := s.c.useAPI.Load()
+	var value N
+	var err error
+	for i := int32(0); i < numAPIs; i++ {
+		value, err = s.call(currentAPI)
+		// This check has a false positive for genuine "not found" errors
+		// when accessing specific objects. The client will then try
+		// all API version until it arrives back at the original one - not
+		// nice because of the overhead, but not wrong semantically
+		// and shouldn't occur often.
+		if apierrors.IsNotFound(err) {
```

Comment:
Is this how "If the apiserver configuration changes" is detected like the commit message describes, based on hitting a NotFound error?